anoncreds/protocol/types.py

```diff
@@ -193,6 +193,35 @@ def __eq__(self, other):
                and self.Z == other.Z and self.seqId == other.seqId \
                and dict(self.R) == dict(other.R)
 
+    def to_str_dict(self):
+        public_key = {
+            'n': str(crypto_int_to_str(self.N)),
+            's': str(crypto_int_to_str(self.S)),
+            'rms': str(crypto_int_to_str(self.Rms)),
+            'rctxt': str(crypto_int_to_str(self.Rctxt)),
+            'z': str(crypto_int_to_str(self.Z)),
+            'r': {}
+        }
+
+        for key in self.R:
+            public_key['r'][key] = str(crypto_int_to_str(self.R[key]))
+
+        return public_key
+
+    @classmethod
+    def from_str_dict(cls, data):
+        N = strToCryptoInteger(data['n'])
+        Rms = strToCryptoInteger(data['rms'] + 'mod' + data['n'])
+        Rctxt = strToCryptoInteger(data['rctxt'] + 'mod' + data['n'])
+        S = strToCryptoInteger(data['s'] + 'mod' + data['n'])
+        Z = strToCryptoInteger(data['z'] + 'mod' + data['n'])
```

Comment:
Please create a method for such serialization:
- a field 'x' and 'n' on input;
strToCryptoInteger(x + 'mod' + n) on output